File: C_Programming/Practise_Programs/PP_031.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_036
//
//  Description       : This program accepts a frequency from the
//                      user and displays numbers from 0 to the
//                      given frequency.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////
//
//  Function Name     : Display()
//
//  Description       : It is used to display numbers from 0 to
//                      the given number. It also checks whether
//                      the given input is negative.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//////////////////////////////////////////////////////////////////

void Display(int iNo)
{
    int iCnt = 0;

    if(iNo < 0)
    {
        printf("Invalid input");
        return;
    }

    for(iCnt = 0; iCnt <= iNo; iCnt++)
    {
        printf("%d", iCnt);
    }
}

int main()
{
    int iValue = 0;

    printf("Enter the frequency : ");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter the frequency : 5
//
//  Sample Output :
//      012345
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_031.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_031
//
//  Description       : This program accepts a number from the user
//                      and counts the number of odd digits present
//                      in the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 10/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class DigitX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CountOddDigits()
    //
    //  Description      : It is used to count the number of odd
    //                     digits present in the given number by
    //                     extracting each digit and checking
    //                     whether it is odd.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 10/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public int CountOddDigits(int iNo)
    {
        int iDigit = 0;
        int iCount = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;

            if(iDigit % 2 != 0)
            {
                iCount++;
            }

            iNo = iNo / 10;
        }

        return iCount;
    }
}

class PP_031
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;
        int iRet = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = dobj.CountOddDigits(iValue);

        System.out.println("Number of Odd Digits are : " + iRet);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter Number :
//      123456
//
//  Sample Output :
//      Number of Odd Digits are : 3
//
//////////////////////////////////////////////////////////////////
